plugins/outputs/cloudwatch/cloudwatch.go

```diff
@@ -571,36 +572,58 @@ func (c *CloudWatch) BuildMetricDatum(point telegraf.Metric) []*cloudwatch.Metri
 // 30 dimensions per metric so we only keep up to the first 30 alphabetically.
 // This always includes the "host" tag if it exists.
 // See https://github.com/aws/amazon-cloudwatch-agent/issues/398
-func BuildDimensions(mTags map[string]string) []*cloudwatch.Dimension {
+func BuildDimensions(mTags map[string]string, globalDimensions map[string]string) []*cloudwatch.Dimension {
 	dimensions := make([]*cloudwatch.Dimension, 0, MaxDimensions)
 
+	hasHostDimension := false
+
 	// This is pretty ugly but we always want to include the "host" tag if it exists.
 	if host, ok := mTags["host"]; ok && host != "" {
+		hasHostDimension = true
 		dimensions = append(dimensions, &cloudwatch.Dimension{
 			Name:  aws.String("host"),
 			Value: aws.String(host),
 		})
 	}
 
-	var keys []string
-	for k := range mTags {
-		if k != "host" {
-			keys = append(keys, k)
+	globalDimensionKeys := make([]string, 0, len(globalDimensions))
+
+	for key, value := range globalDimensions {
+		if key != "" && value != "" {
+			globalDimensionKeys = append(globalDimensionKeys, key)
 		}
 	}
-	sort.Strings(keys)
 
-	for _, k := range keys {
-		if len(dimensions) >= MaxDimensions {
-			log.Printf("D! max MaxDimensions %v is less than than number of dimensions %v thus only taking the max number", MaxDimensions, len(dimensions))
-			break
-		}
+	if len(globalDimensionKeys)+len(dimensions) > MaxDimensions {
+		log.Printf("D! max MaxDimensions %v is less than than number of global dimensions plus host dimension (if present) %v thus only taking the max number", MaxDimensions, len(globalDimensionKeys)+len(dimensions))
+		sort.Strings(globalDimensionKeys)
+		globalDimensionKeys = globalDimensionKeys[:MaxDimensions-len(dimensions)]
+	}
 
-		value := mTags[k]
-		if value == "" {
+	for _, key := range globalDimensionKeys {
+		if hasHostDimension && key == "host" {
```

Comment:
This would be an interesting use case... Do you think we need any guard against configuring `host` in the global dimensions map?